app/javascript/controllers/file_auth_controller.js

```diff
@@ -264,4 +271,42 @@ export default class extends Controller {
   hideMessagePanel() {
     this.messagePanelTarget.hidden = true
   }
+
+  // To see if item is restricted by location, check the probe service json response
+  handleLocationRestricted(json, accessService) {
+    // The probe auth service is called for each file separately.
+    // If the location restriction target is available, then trigger auth denied message.
+    // The event will lead to the locked icon being displayed for this item
+    // This allows the lock window to show
+    const event = new CustomEvent('auth-denied', { accessService: accessService })
+    window.dispatchEvent(event)
+    // Make the location restricted banner visible
+    this.locationRestrictionTarget.hidden = false
+    // Display the location restriction message based on the authorization response
+    this.locationRestrictionMessageTarget.innerHTML = this.retrieveRestrictedLocationMessage(json)
+
+  }
+
+  // Checks the result of the probe auth request to see if access is restricted to location
+  // This code depends on the text returned by probe service, so changes to the heading
+  // should be reflected here as well.
+  isLocationRestricted(json) {
+    if(json.status == '401' && 'heading' in json && 'en' in json.heading && json.heading.en.length
+      && json.heading.en[0].startsWith('Content is restricted to location'))
+      return true
+    return false
+  }
+
+  // Extract just the location from the message
+  extractRestrictedLocation(accessMessage) {
+    const probeServicePrefix = 'Content is restricted to location '
+    if(accessMessage != null && accessMessage.startsWith(probeServicePrefix))
+      return accessMessage.substring(probeServicePrefix.length, accessMessage.length)
+
+    return 'site visitors to the Stanford libraries'
+  }
+
+  retrieveRestrictedLocationMessage(json) {
+    return 'Access is restricted to the ' + this.extractRestrictedLocation(json.heading.en[0]) + '. See Access conditions for more information.'
```

Comment:
Can this whole first sentence move to the IIIF Auth response heading in Stacks?

Reply:
I am working on that in a stacks PR, so I agree that would be better

Reply:
For now, so we can get this in because it is a blocker, can we just return what we are getting from stacks since it is very close to what we want and we can have the language update done after this is in?

Reply:
The other PR is merged so we should be good to go with the updated strings